kv/util/runtime/value.go

```diff
@@ -37,24 +38,34 @@ var (
 
 // Value is a value that can be updated during runtime.
 type Value interface {
-	// Key is the key associated with value.
-	Key() string
-
 	// Watch starts watching for value updates.
 	Watch() error
 
 	// Unwatch stops watching for value updates.
 	Unwatch()
 }
 
-// UnmarshalFn unmarshals a kv value and extracts its payload.
-type UnmarshalFn func(value kv.Value) (interface{}, error)
+// Updatable can be updated.
+// TODO: Move to m3x/watch.
+type Updatable interface {
+	close.SimpleCloser
+
+	// C returns the notification channel for updates.
+	C() <-chan struct{}
+}
+
+// NewUpdatableFn creates an updatable.
+type NewUpdatableFn func() (Updatable, error)
+
+// GetFn returns the latest value.
+type GetFn func(updatable Updatable) (interface{}, error)
 
 // ProcessFn processes a value.
+// If the value is versionable, then only newer value will be processed.
 type ProcessFn func(value interface{}) error
 
-// updateWithLockFn updates a value while holding a lock.
-type updateWithLockFn func(value kv.Value) error
+// processWithLockFn updates a value while holding a lock.
+type processWithLockFn func(value interface{}) error
```

Comment:
Hm I actually think `updateWithLockFn` is a better name here. Also the param should be a `Versionable` I'd think?